### PR TITLE
Feature/vocascan cli

### PR DESCRIPTION
**Pull request by luwol03**
| Status  | Type  | Env Vars Change |
| :---: | :---: | :---: |
| :white_check_mark: Ready | Feature/Refactor | No |

## Description

This Pull Request adds a CLI to manage things. I'll make the documentation for the commands ASAP. But in short, the cli can handle following things:
- start vocascan-server
- create, read, update, delete users
- generate secrets

To develop and test this locally you can do a `npm link` in the root of the repo, so you are able to use the `vocascan-server` command. There is also a automatically generated help command from command. (`vocascan-server --help`)

## Motivation and Context

I refactored that because now its very easy to run a vocascan server also programatically, which would help us in the future by maybe integrating a server into the desktop app to use it like a offline/no sync variant. 

```js
const runServer = require('vocascan-server');

(async () => {
  const server = await runServer({
    server: {
      port: 8989,
    },
  });

  setTimeout(() => {
    server.close();
  }, 2000);
})();
```

## Checklist

- [x] I have read the **CONTRIBUTING** document.
- [x] I have considered the accessibility of my changes (i.e. did I add proper content descriptions to images, or run my changes with talkback enabled?)
- [x] I have documented my code if needed

### CODE REVIEW

File: database/Migration.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      .then(() => {
+        logger.info(chalk`{green ✓} ${this.displayName} done in {cyan (${round(performance.now() - startTime, 3)}ms)}`);
+      })
+      .catch((err) => {
+        logger.error(`An Error occurred while ${this.displayName}`);
+        throw err;
+      });
+  }
+
+  // TODO: add down functionality
```
Comment (noctera) [inline database/Migration.js:58]: Is this intended to be done in this PR?

Comment (luwol03) [reply database/Migration.js:58]: No. Its just a comment to do that sometime in the future.

File: cmd/generate.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const chalk = require('chalk');
+const { generateRandomString } = require('../app/utils');
+
+const generate = new commander.Command('generate');
+
+const SECRET_TYPES = {
+  JWT_SECRET: 'JWT_SECRET',
+};
+
+generate.description('generate secrets to use in the config file');
```
Comment (noctera) [inline cmd/generate.js]: This description is duplicate. You should use a more general description.

Comment (luwol03) [reply cmd/generate.js]: Do you have a better idea? Something like `generate things`? But that doesnt sound good.

Comment (noctera) [reply cmd/generate.js]: maybe ``generation functions for strings needed for config files``

**Review by noctera — APPROVED** [review]

**Pull request merged**